docs/topics/native/native-spm.md

```diff
@@ -0,0 +1,194 @@
+            checksum:"<checksum calculated for the ZIP file>"
+      ]
+   )
+   ```
+8. You can check if the package manifest is valid by running the following command next to the `Package.swift` file:
+
+    ```shell
+    swift package describe
+    ```
+    
+    The output will describe any found errors, or show the parsed tree of the manifest if it is grammatically correct. 
+9. Push the `Package.swift` file to the repository you settled on earlier. Make sure to create a git tag with the
+Semantic Version of the package.
+
+Now that both files are accessible, you can try and set up the dependency in Xcode:
+* In an Xcode project, choose **File | Add Package Dependencies...** and provide the Git URL for the repo with
+the `Package.swift` file.
+* In an SPM project, create a dependency following the Apple documentation on [Package.Dependency](https://developer.apple.com/documentation/packagedescription/package/dependency).
+
+## Exporting different modules as a single XCFramework
+
+If your project contains unrelated KMP modules which you would like to export as a single iOS binary, you can create
+an umbrella module and combine other modules in it.
+
+For example, you have a `network` and a `database` module, which you combine in an `together` module.
+
+1. In the `together/build.gradle.kts` file, specify dependencies and the framework configuration:
+
+    ```kotlin
+    kotlin {
+        val frameworkName = "together"
+        val xcf = XCFramework()
+    
+        listOf(
+            iosX64(),
+            iosArm64(),
+            iosSimulatorArm64()
+        ).forEach { iosTarget ->
+            // Same as in the example above, with added export calls for dependencies
+            iosTarget.binaries.framework {
+                export(projects.network)
+                export(projects.database)
+    
+                baseName = frameworkName
+                xcf.add(this)
+            }
+        }
+    
+        // Dependencies set as api to export underlying modules
+        sourceSets {
+            commonMain.dependencies {
+                api(projects.network)
+                api(projects.database)
+            }
+        }
+    }
+    ```
+
+2. Each of the included modules should have its iOS targets configured, for example:
+
+    ```kotlin
+    kotlin {
+        // ...
+        listOf(
+            iosX64(),
+            iosArm64(),
+            iosSimulatorArm64()
+        ).forEach { iosTarget ->
+            iosTarget.binaries.framework {
+                baseName = "network"
+                isStatic = true
+            }
+        }
+        //...
+    }
+    ```
+
+3. Currently, a framework cannot be assembled if the module being exported does not contain any source code. To work around
+this:
+   1. Create a source file inside the `together` folder, for example, `together/src/commonMain/kotlin/Together.kt`.
+   2. Create an empty `main()` function inside this file:
+       ```kotlin
+      fun main() {
```

Comment:
It should be enough to create an empty `stub.kt` file. Let's not ask to write any code in this stub

Reply:
It didn't work for me with an empty file, but now it does. Good news!